packages/sanity-plugin-nacelle-input/src/hooks/useHailFrequency.js

```diff
@@ -0,0 +1,97 @@
+import { useState, useEffect } from 'react'
+
+import useSWR from 'swr'
+import fetch from 'isomorphic-unfetch'
+import config from 'config:@nacelle/sanity-plugin-nacelle-input'
+
+async function fetchFromHailFrequency({
+  query,
+  first,
+  spaceToken,
+  endpoint,
+  searchTerm
+}) {
+  const variables = searchTerm
+    ? {
+        first,
+        searchFilter: {
+          fields: ['HANDLE', 'TITLE'],
+          term: searchTerm
+        }
+      }
+    : {
+        first
+      }
+  return await fetch(endpoint, {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+      'x-nacelle-space-token': spaceToken
+    },
+    body: JSON.stringify({
+      query,
+      variables
+    })
+  }).then((res) => res.json())
+}
+
+async function fetcher(query, spaceToken, endpoint, type, searchTerm) {
+  let data = []
+
+  const res = await fetchFromHailFrequency({
+    query,
+    first: 200,
```

Comment:
We might be able to fetch less data here? The search box only has a handful of options in frame at any given time. Was at 500, but cut it to 200 since we aren't doing any in-memory filtering so having too many extra values isn't super useful.